fmskill/model/factory.py

```diff
@@ -1,11 +1,60 @@
 import os
+from pathlib import Path
+from typing import Literal, Optional
 import pandas as pd
+import xarray as xr
+
+from fmskill import model, types, utils
+import mikeio
 
 from .dfs import DataArrayModelResultItem, DfsModelResultItem
 from ._pandas import DataFramePointModelResultItem, DataFrameTrackModelResultItem
 from ._xarray import XArrayModelResultItem
 
 
+type_lookup = {
+    "point": model.PointModelResult,
+    "track": model.TrackModelResult,
+    "unstructured": model.DfsuModelResult,
+    "grid": model.GridModelResult,
+}
+
+
+class ModelResult_new:
+    def __new__(
+        cls,
+        data: types.DataInputType,
+        model_type: Optional[Literal["point", "track", "unstructured", "grid"]] = None,
```

Comment:
I suppose `model_type` could be also be enum:
```python
from enum import Enum, auto

class ModelType(Enum):
    Point = auto(),
    Track = auto(),
    Unstructured = auto(),
    Grid = auto()
    
    def __str__(self) -> str:
        return self.name.lower()
```

```python
>>> ModelType.Grid
<ModelType.Grid: 4>
>>> str(ModelType.Unstructured)
'unstructured'
```

What do you think about that?

Reply:
I like Enum

Reply:
Would we want the user to use that enum or just use it internally?

Reply:
I would say internally.